### PR TITLE
Feature sw cleaning

### PR DESCRIPTION
**Pull request by miguelmaso**
In this PR I introduce the next changes in shallow water application:
- some coding convention and doxygen
- elements derive from a base element (PrimitiveVarElement)

### CODE REVIEW

**Comment by miguelmaso** [comment]
@pooyan-dadvand The Quality Review is showing two issues.
The first issue is about a struct variable declared on a header. The variable will be used in the source file, as the other variables. Have I done an mistake or is it a reviewer problem?

**Comment by pooyan-dadvand** [comment]
Where are you using that variables? I could not find it neither

**Comment by miguelmaso** [comment]
I am using this variables in the .cpp files
But I don't understand why some variables are passing the test and other variables are failing

**Comment by pooyan-dadvand** [comment]
I don't know... Meanwhile, what I don't like is that you are not initializing them in the constructor....

File: applications/ShallowWaterApplication/custom_elements/conserved_var_element.hpp [hunk 1]
```diff
@@ -23,84 +23,266 @@
...
+        return boost::make_shared< ConservedVarElement < TNumNodes > >(NewId, this->GetGeometry().Create(rThisNodes), pProperties);
+        KRATOS_CATCH("")
+    }
 
-    Element::Pointer Create(IndexType NewId, NodesArrayType const& ThisNodes,  PropertiesType::Pointer pProperties) const
+    Element::Pointer Create(IndexType NewId, GeometryPointerType pGeom, PropertiesPointerType pProperties) const override
     {
         KRATOS_TRY
-        return Element::Pointer(new ConservedVarElement(NewId, GetGeometry().Create(ThisNodes), pProperties));
+        return boost::make_shared< ConservedVarElement < TNumNodes > >(NewId, pGeom, pProperties);
```
Comment (loumalouomega) [inline applications/ShallowWaterApplication/custom_elements/conserved_var_element.hpp]: Change boost:: for Kratos::

File: applications/ShallowWaterApplication/custom_elements/euler_cons_var_element.cpp [hunk 2]
```diff
@@ -81,318 +124,258 @@ namespace Kratos
...
+        // Initialize outputs
+        rVariables.height = 0;
+        rVariables.momentum = ZeroVector(2);
+        rVariables.velocity = ZeroVector(2);
+        rVariables.velocity_grad = ZeroMatrix(2,2);
+
+        // check if the element is close to dry
         bool near_dry = false;
-        
+        double threshold = 1e-1;
```
Comment (loumalouomega) [inline applications/ShallowWaterApplication/custom_elements/euler_cons_var_element.cpp:253]: A little bit big?

**Review by loumalouomega — CHANGES_REQUESTED** [review]
Minor changes requested

File: applications/ShallowWaterApplication/custom_elements/conserved_var_element.hpp (hunk 1 above)
Comment (miguelmaso) [reply applications/ShallowWaterApplication/custom_elements/conserved_var_element.hpp]: I am getting an error with Kratos::make_shared

Comment (miguelmaso) [reply applications/ShallowWaterApplication/custom_elements/conserved_var_element.hpp]: changed

**Review by loumalouomega — APPROVED** [review]
Okidoki

**Pull request merged**